File: CS/InplageGaugeControl/Form1.cs
using DevExpress.XtraEditors;
using DevExpress.XtraGauges.Base;
using DevExpress.XtraGauges.Presets.Styles;
using DevExpress.XtraGauges.Win;
using DevExpress.XtraGauges.Win.Gauges.Circular;

namespace InplaceGaugeControl {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            
            CreateGaugeFromCode(Theme.MagicLight);

            for(int i = 25; i < 75; i += 5)
                dataSet.Tables[0].Rows.Add(new object[] { i });
        }
        GaugeControl gaugeControl;
        CircularGauge circularGauge;
        void CreateGaugeFromCode(Theme theme) {
            gaugeControl = new GaugeControl();
            gaugeControl.Size = new System.Drawing.Size(200, 200);

            circularGauge = gaugeControl.AddCircularGauge();
            circularGauge.AddDefaultElements();

            var style = StyleLoader.Load("Circular", theme.Name, "Full");
            if(style != null)
                style.Apply(circularGauge);
        }
        void gridView1_CustomUnboundColumnData(object sender, DevExpress.XtraGrid.Views.Base.CustomColumnDataEventArgs e) {
            if(e.Column == colGauge && e.IsGetData) {
                int speed = (int)gridView.GetRowCellValue(e.ListSourceRowIndex, colSpeed);
                circularGauge.Scales[0].Value = (float)speed;
                e.Value = ((IGaugeContainer)gaugeControl).GetImage(200, 200);
            }
        }
    }
}

File: VB/InplageGaugeControl/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGauges.Base
Imports DevExpress.XtraGauges.Presets.Styles
Imports DevExpress.XtraGauges.Win
Imports DevExpress.XtraGauges.Win.Gauges.Circular

Namespace InplaceGaugeControl
	Partial Public Class Form1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()

			CreateGaugeFromCode(Theme.MagicLight)

			For i As Integer = 25 To 74 Step 5
				dataSet.Tables(0).Rows.Add(New Object() { i })
			Next i
		End Sub
		Private gaugeControl As GaugeControl
		Private circularGauge As CircularGauge
		Private Sub CreateGaugeFromCode(ByVal theme As Theme)
			gaugeControl = New GaugeControl()
			gaugeControl.Size = New System.Drawing.Size(200, 200)

			circularGauge = gaugeControl.AddCircularGauge()
			circularGauge.AddDefaultElements()

			Dim style = StyleLoader.Load("Circular", theme.Name, "Full")
			If style IsNot Nothing Then
				style.Apply(circularGauge)
			End If
		End Sub
		Private Sub gridView1_CustomUnboundColumnData(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.CustomColumnDataEventArgs) Handles gridView.CustomUnboundColumnData
			If e.Column Is colGauge AndAlso e.IsGetData Then
				Dim speed As Integer = CInt(Fix(gridView.GetRowCellValue(e.ListSourceRowIndex, colSpeed)))
				circularGauge.Scales(0).Value = CSng(speed)
				e.Value = (CType(gaugeControl, IGaugeContainer)).GetImage(200, 200)
			End If
		End Sub
	End Class
End Namespace